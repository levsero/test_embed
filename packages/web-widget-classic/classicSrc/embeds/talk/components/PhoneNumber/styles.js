import styled from 'styled-components'
import { zdColorBlue600 } from '@zendeskgarden/css-variables'

const PhoneLink = styled.a`
  color: ${zdColorBlue600};
  text-decoration: underline;

  &:hover,
  &:focus,
  &:active {
    color: ${zdColorBlue600};
  }
`

export { PhoneLink }
