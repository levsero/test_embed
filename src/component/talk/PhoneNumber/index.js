import React from 'react';
import { locals as styles } from './styles/index.scss';
import PropTypes from 'prop-types';

const PhoneNumber = ({ phoneNumber, formattedPhoneNumber }) => {
  return (
    <a className={styles.phoneLink} href={`tel:${phoneNumber}`} target='_blank'>
      {formattedPhoneNumber}
    </a>
  );
};

PhoneNumber.propTypes = {
  phoneNumber: PropTypes.string.isRequired,
  formattedPhoneNumber: PropTypes.string.isRequired,
};

export default PhoneNumber;
