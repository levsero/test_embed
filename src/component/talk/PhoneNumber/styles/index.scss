@import '../../../../styles/_vars.scss';

.phoneLink {
  color: $zd-color-blue-600;
  text-decoration: underline;

  &:hover, &:focus, &:active {
    color: $zd-color-blue-600;
  }
}
