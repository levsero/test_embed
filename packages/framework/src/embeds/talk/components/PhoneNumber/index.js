import PropTypes from 'prop-types'
import { PhoneLink } from './styles'
import { TEST_IDS } from 'src/constants/shared'

const PhoneNumber = ({ phoneNumber, formattedPhoneNumber }) => (
  <PhoneLink href={`tel:${phoneNumber}`} target="_blank" data-testid={TEST_IDS.TALK_PHONE_NUMBER}>
    {formattedPhoneNumber}
  </PhoneLink>
)

PhoneNumber.propTypes = {
  phoneNumber: PropTypes.string.isRequired,
  formattedPhoneNumber: PropTypes.string.isRequired,
}

export default PhoneNumber
