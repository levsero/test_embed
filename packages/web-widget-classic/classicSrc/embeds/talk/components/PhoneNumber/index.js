import { TEST_IDS } from 'classicSrc/constants/shared'
import PropTypes from 'prop-types'
import { PhoneLink } from './styles'

const PhoneNumber = ({ phoneNumber, formattedPhoneNumber }) => (
  <PhoneLink
    href={`tel:${phoneNumber}`}
    target="_blank"
    data-testid={TEST_IDS.TALK_PHONE_NUMBER}
    dir="ltr"
  >
    {formattedPhoneNumber}
  </PhoneLink>
)

PhoneNumber.propTypes = {
  phoneNumber: PropTypes.string.isRequired,
  formattedPhoneNumber: PropTypes.string.isRequired,
}

export default PhoneNumber
