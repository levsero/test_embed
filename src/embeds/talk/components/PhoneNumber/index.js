import React from 'react'
import PropTypes from 'prop-types'
import { PhoneLink } from './styles'

const PhoneNumber = ({ phoneNumber, formattedPhoneNumber }) => {
  return (
    <PhoneLink href={`tel:${phoneNumber}`} target="_blank">
      {formattedPhoneNumber}
    </PhoneLink>
  )
}

PhoneNumber.propTypes = {
  phoneNumber: PropTypes.string.isRequired,
  formattedPhoneNumber: PropTypes.string.isRequired
}

export default PhoneNumber
