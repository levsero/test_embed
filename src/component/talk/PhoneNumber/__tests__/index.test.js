import { render } from '@testing-library/react'
import React from 'react'

import PhoneNumber from '../'

test('renders the component', () => {
  const { container } = render(
    <PhoneNumber phoneNumber={'95556666'} formattedPhoneNumber={'9-555-6666'} />
  )

  expect(container).toMatchSnapshot()
})
