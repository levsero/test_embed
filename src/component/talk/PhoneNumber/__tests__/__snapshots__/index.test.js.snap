// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div>
  <a
    class="phoneLink"
    href="tel:95556666"
    target="_blank"
  >
    9-555-6666
  </a>
</div>
`;
